packages/m2m-gateway/src/utils/validators/attributeValidators.ts

```diff
@@ -8,10 +11,21 @@ export function assertAttributeKindIs<
   K extends attributeRegistryApi.AttributeKind
 >(
   attribute: attributeRegistryApi.Attribute,
-  expectedKind: K
+  expectedKind: K,
+  error: Extract<
+    ErrorCodes,
+    "unexpectedAttributeKind" | "attributeNotFound"
+  > = "unexpectedAttributeKind"
 ): asserts attribute is attributeRegistryApi.Attribute & { kind: K } {
   if (attribute.kind !== expectedKind) {
-    throw unexpectedAttributeKind(attribute);
+    match(error)
+      .with("unexpectedAttributeKind", () => {
+        throw unexpectedAttributeKind(attribute);
+      })
+      .with("attributeNotFound", () => {
+        throw attributeNotFound(attribute);
+      })
+      .exhaustive();
   }
```

Comment:
This change looks unrelated; maybe it's a merge issue. Revert to how it's now merged in dev